src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegisterFailedException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final User user = new User("Bob_Super", "Bob12345", 18);
+
+    private static RegistrationServiceImpl service;
+    private static StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao.add(user);
+    }
+
+    @Test
+    void register_validUser_Ok() {
+        User validNewUser = new User("_Emily_", "_emily_", 22);
+
+        User validUserResult = service.register(validNewUser);
+
+        assertEquals(validUserResult, validNewUser);
+    }
+    
+    @Test
+    void register_validEdgeValuesUser_Ok() {
+        User validEdgeNewUser = new User("_Emily", "emily_", 18);
+
+        User validEdgeLoginResult = service.register(validEdgeNewUser);
+
+        assertEquals(validEdgeLoginResult, validEdgeNewUser);
+    }
+
+    @Test
+    void register_sameLoginUser_NotOk() {
+        User sameLoginUser = new User("Bob_Super", "Bob2345", 20);
+
+        assertThrows(RegisterFailedException.class, () ->
+                service.register(sameLoginUser));
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(IllegalArgumentException.class, () ->
+                service.register(null));
+    }
+
+    @Test
+    void register_wrongLogin_NotOk() {
```

Comment:
Let's have more informative method names. See everywhere
```suggestion
    void register_loginLengthLessThanMinimalRequired_NotOk() {
```